packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadView/UploadView.tsx

```diff
@@ -8,43 +8,28 @@ import { ControlsContextProvider } from '../../../controls/context';
 import { CLASS_BASE } from '../../constants';
 import { Controls } from '../../Controls';
 import { Title } from '../Controls/Title';
-import { ActionStartControl } from '../../../controls/ActionStartControl';
 import { ActionCancelControl } from '../../../controls/ActionCancelControl';
+import { ActionExitControl } from '../../../controls/ActionExitControl';
+import { ActionStartControl } from '../../../controls/ActionStartControl';
 import { DataTableControl } from '../../../controls/DataTableControl';
 import { DropZoneControl } from '../../../controls/DropZoneControl';
-import { getFileSelectionType } from './getFileSelectionType';
-import { useStore } from '../../../providers/store';
 import { resolveClassName } from '../../utils';
 import { getActionViewTableData } from '../getActionViewTableData';
+import { useDisplayText } from '../../../displayText';
+
 import { useUploadView } from './useUploadView';
 import { UploadViewProps } from './types';
 
-const { Exit, Overwrite } = Controls;
+const { Overwrite } = Controls;
 
 export const ICON_CLASS = `${CLASS_BASE}__action-status`;
 
-export const UploadView = ({
+export function UploadView({
   className,
-  onExit: onExitProps,
-}: UploadViewProps): React.JSX.Element => {
-  const [{ actionType, files }, dispatchStoreAction] = useStore();
-  // launch native file picker on intiial render if no files are currently in state
-  const selectionTypeRef = React.useRef<'FILE' | 'FOLDER' | undefined>(
-    getFileSelectionType(actionType, files)
-  );
-
-  React.useEffect(() => {
-    const selectionType = selectionTypeRef.current;
-    if (!selectionType) {
-      return;
-    }
-
-    dispatchStoreAction({ type: 'SELECT_FILES', selectionType });
-
-    return () => {
-      selectionTypeRef.current = undefined;
-    };
-  }, [dispatchStoreAction]);
```

Comment:
Launch of the native picker on upload files/folder action select is no longer in scope due to friction with routed usage of `StorageBrowser`